.github/workflows/test.yml

```diff
@@ -158,6 +158,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "${{ matrix.python }}"
+          cache: pip
```

Comment:
Do you know after when / how the cache will be invalidated by default?

Reply:
by default the cache key is the hash of requirements.txt, so if our dependencies change, this will clear the cache. We can set [cache-dependency-path](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages) to be more precise, which I've done now.

Note that an outdated cache doesn't cause problems because it only caches pip's own cache dir, i.e. downloaded wheels, not install results. So when there is a cache hit on an outdated package, pip will still lookup and install the latest available version, updating the cache.

Reply:
Ooooh then im very much in favor of this without any hesitation!